### DISCUSSION TITLE
Building a .jimage with spork dependency

### DISCUSSION THREAD

**Discussion by kredati** [janet-lang/janet#1453] [Q&A] [answered]
Hello, friends! I'll be sure to show off what I'm working on when it's ready (and it's just about ready). For now, I'm trying to figure out how to package up the project. 

For that, I need to build an image of a janet file that has exactly one dependency: spork's `json/encode`. I can't figure out how to build this successfully.

In the file: I`(import spork/json :as j)`. Everything works as it should in the repl and when evaluated as a janet source file.

But `janet -c src/myscript.janet build/myscript.jimage` and I get the following:
```
error: no registry value and cannot marshal <cfunction json/decode>
  in marshal [src/core/marsh.c] on line 1638
  in make-image [boot.janet] on line 2757, column 3
  in c-switch [boot.janet] (tailcall) on line 4066, column 36
  in cli-main [boot.janet] on line 4102, column 13
```

I also tried to build a `spork.jimage` file with the following code:
```
(def env (require "spork/json"))
(spit "spork.jimage" (make-image env))
```
And get the identical error. (And a slightly different one if I `require` the whole "spork".)

This seems like something `jpm` could do? But I can't see how to get an image out of `jpm`.

Help is extremely appreciated! (Especially because the alternative is writing my own JSON encoder in pure Janet, which is, like, not exactly my idea of a good time.)

**Comment by CFiggers** [comment, marked as answer]
I ran into this exact issue and to solve it I ended up making a pure-Janet library for JSON encoding and decoding called [Jayson](https://github.com/CFiggers/jayson). The API is (almost) identical to spork/json, with the one exception that Jayson uses `:json/null` to represent JSON Null rather than spork/json's `:null`. Because it's pure Janet, it works with .jimage files no problem.

Feel free to give Jayson a try and see if it solves your problem—I'm also interested to know if anyone else has a solution for this problem that I didn't think of. 🙂

**Reply by kredati** [reply]
Thank you so much. I literally had the wikipedia page for JSON open to start unpicking this particular knot.

**Reply by sogaiu** [reply]
On the note of pure Janet JSON implementations, FWIW, there is [this list](https://github.com/sogaiu/janet-jsonish?#pure-janet).

**Reply by ghost** [reply]
Hi Jayson!
